flink/src/test/java/org/apache/iceberg/flink/TestIcebergConnector.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.iceberg.flink;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import org.apache.flink.configuration.CoreOptions;
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CloseableIterator;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class TestIcebergConnector extends FlinkTestBase {
+
+  private static TableEnvironment tEnv;
+
+  @Rule
+  public final TemporaryFolder warehouse = new TemporaryFolder();
+
+
+  @BeforeClass
+  public static void beforeClass() {
+    EnvironmentSettings settings = EnvironmentSettings
+        .newInstance()
+        .useBlinkPlanner()
+        .inBatchMode()
+        .build();
+
+    tEnv = TableEnvironment.create(settings);
+    tEnv.getConfig().getConfiguration()
+        .set(CoreOptions.DEFAULT_PARALLELISM, 1)
+        .set(FlinkTableOptions.TABLE_EXEC_ICEBERG_INFER_SOURCE_PARALLELISM, false);
+  }
+
+  @Test
+  public void testHadoopCatalog() {
+    Map<String, String> tableProps = Maps.newHashMap();
+    tableProps.put("connector", "iceberg");
+    tableProps.put("catalog-name", "test-hadoop");
+    tableProps.put("catalog-type", "hadoop");
+    tableProps.put("catalog-database", "local_db");
+    tableProps.put(CatalogProperties.WAREHOUSE_LOCATION, warehouseRoot());
+
+    sql("CREATE TABLE hadoop_table (id BIGINT, data STRING) WITH %s", toWithClause(tableProps));
```

Comment:
Here we configure the `catalog-name` and `catalog-database` in the props. Flink SQL supports the namespace syntax. trying to understand if that is a problem.

https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/dev/table/sql/create/
```
CREATE TABLE  [catalog_name.][db_name.]table_name
```

Reply:
Here I'd like to explain that why I have planned to introduce the `catalog-name` & `catalog-database` in the table properties:   For flink users,   the SQL `CREATE TABLE iceberg_sink(id BIGINT, data STRING) WITH ('connector'='iceberg', ... ) ` is actually creating a flink table which is managed in the flink's in-memory catalog `default_catalog` and `default_database`,  it don't have any relationship to the iceberg catalog which manage the storage layer's database -> table -> table location and provide global table lock services.  

Those table properties from flink create DDL is actually building the mapping relationship between flink's in-memory catalog and the storage layer catalog.

If we use the `CREATE TABLE [catalog_name.][db_name]table_name`, that's what the current way we've provided in the [document](https://iceberg.apache.org/flink/), I mean the flink's catalog is exactly to the same iceberg catalog.

Reply:
understand the meaning of specifying catalog database, but I still don't understand why catalog name must be specified What's the use of specifying catalog name? For Flink, any value of catalog name can be specified, which has no impact on my data processing tasks. @openinx